src/read.c

```diff
@@ -1046,13 +1046,21 @@ static void
 seffect_enchant_armor(struct obj **sobjp)
 {
     struct obj *sobj = *sobjp;
+    int otyp = sobj->otyp;
     register schar s;
     boolean special_armor;
     boolean same_color;
     boolean sblessed = sobj->blessed;
     boolean scursed = sobj->cursed;
     boolean confused = (Confusion != 0);
     boolean old_erodeproof, new_erodeproof;
+    boolean already_known = (sobj->oclass == SPBOOK_CLASS /* spell */
+                             || objects[otyp].oc_name_known);
```

Comment:
For NetHack code style consistency, you're also going to want to move the `struct obj *otmp;` declaration up here separate from its initialization.